packages/application-generic/src/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -37,7 +37,9 @@ export class InMemoryProviderService {
   public inMemoryProviderClient: InMemoryProviderClient;
   public inMemoryProviderConfig: InMemoryProviderConfig;
 
-  constructor(private enableAutoPipelining?: boolean) {
+  constructor(private enableAutoPipelining?: boolean) {}
+
+  public initialize(): void {
```

Comment:
Preparation to make this async when we need to plug in the GetFeatureFlag use case to retrieve the global feature flag to check if the cluster mode is enabled.